### ISSUE TITLE
x/sys/unix: add support for vectorized ("scatter/gather") IO syscalls on Darwin

### ISSUE THREAD

**Issue by mschoenlaub** [golang/go#64710]
### Proposal Details

The following syscalls could be supported on modern OSX versions (Darwin 11 and newer)

- `unix.Readv(fd int, iovs [][]byte) (n int, err error)`
- `unix.Preadv(fd int, iovs [][]byte, offset int64) (n int, err error)` 
- `unix.Writev(fd int, iovs [][]byte) (n int, err error)`
- `unix.Pwritev(fd int, iovs [][]byte, offset int64) (n int, err error)`

I think https://go-review.googlesource.com/c/sys/+/548795 contains everything needed to implement this on Darwin. 
In cases where the version is too old, the public functions in `unix` would simply return `ENOSYS` instead of running into a `dyld` error due to trying to jump to a function that doesn't exist.

**Comment by adonovan** [comment]
Why just Darwin? p{read,write}v have been supported on [Linux](https://man7.org/linux/man-pages/man2/readv.2.html) and various [BSDs](https://man.freebsd.org/cgi/man.cgi?query=preadv&sektion=2&n=1) for a while. Ideally the sys interface would have implementations for all platforms that support it, even if the macOS implementation needs an additional compatibility check.

**Comment by ianlancetaylor** [comment]
Yes, there is no need for a proposal to add new system calls for a GOOS that are the same as existing system calls for a different GOOS.

**Comment by wwqgtxx** [comment]
By consulting the official header file of APPLE: 

https://github.com/apple-oss-distributions/xnu/blob/a1e26a70f38d1d7daa7b49b258e2f8538ad81650/bsd/sys/uio.h#L269-L280

readv/writev does not have a clear version requirement (according to the commit history, the system call has existed at least in xnu-123.5, which should correspond to macos10.0)

And the version requirement of preadv/pwritev should also be 10.16 instead of 11.0, so the system version check here is wrong and unnecessary.

In addition, the current `darwinKernelVersionMin` implementation does not have any result caching mechanism, which will lead to high operating overhead in high-frequency calls (each call needs to bear 5 sysctl calls brought by `Uname`, and result string parsing).
